### PR TITLE
General cumprod

### LINKED ISSUE #2395: cumprod

### PR DESCRIPTION
**Pull request by kunwarsaaim**
Resolves #2395

### CODE REVIEW

**Comment by RitujaPawas** [comment]
Please add docstring and docstring examples to `ivy/array/general.py` as well as `ivy/container/general.py`

**Comment by kunwarsaaim** [comment]
> Please add docstring and docstring examples to `ivy/array/general.py` as well as `ivy/container/general.py`

@RitujaPawas There is no `cumprod` function in these files.

**Comment by RitujaPawas** [comment]
@kunwarsaaim you need to create cumprod function inside array and container classes, write their docstring and docstring-examples!
 You can look into other functions, which have already been implemented, as a reference! 
Let me know if you need any help!

**Comment by kunwarsaaim** [comment]
> @kunwarsaaim you need to create cumprod function inside array and container classes, write their docstring and docstring-examples! You can look into other functions, which have already been implemented, as a reference! Let me know if you need any help!

@RitujaPawas I have implemented the functions, please have a look.

File: ivy/functional/ivy/general.py [hunk 1]
```diff
@@ -2431,48 +2430,62 @@ def cumprod(
...
+    >>> ivy.cumprod(x, axis=1, exclusive=True, out=y)
+    >>> print(y)
+    {
+        a: ivy.array([[1, 2],
+                      [1, 5],
+                      [1, 11]]),
+        b: ivy.array([[1, 3],
+                      [1, 4],
+                      [1, 5]])
+    }
```
Comment (RitujaPawas) [inline ivy/functional/ivy/general.py:2488]: You need to implement Array instance method and Container instance method as well in the examples!

**Review by RitujaPawas — CHANGES_REQUESTED** [review]
Everything else looks good. Adding instance method examples in ivy/general.py should be enough!

File: ivy/functional/ivy/general.py (hunk 1 above)
Comment (kunwarsaaim) [reply ivy/functional/ivy/general.py:2488]: @RitujaPawas In the deep dive docstrings examples, it is mentioned that array instance methods and container instance methods should be added to `ivy.Array` and `ivy.Container` class only.

**Comment by kunwarsaaim** [comment]
@RitujaPawas Hi, can you have a look at the PR, I have made the requested changes.

**Comment by RitujaPawas** [comment]
Hey.. sorry for the delay. I will review your PR ASAP!

**Review by RitujaPawas — CHANGES_REQUESTED** [review]
HI @kunwarsaaim  the docstring tests are failing because of your examples. Please fix those. and make sure they pass.

**Review by RitujaPawas — APPROVED** [review]
Hey @kunwarsaaim lgtm! thankyou for your contributions!

**Pull request merged**